code/cfile/cfile.cpp

```diff
@@ -634,13 +635,14 @@ void cf_create_directory( int dir_type )
 
 	} while( current_dir != CF_TYPE_ROOT );
 
-	
 	int i;
 
 	for (i=num_dirs-1; i>=0; i-- )	{
 		cf_create_default_path_string( longname, sizeof(longname)-1, dir_tree[i], NULL );
-		mprintf(( "CFILE: Creating new directory '%s'\n", longname ));
-        mkdir_recursive(longname);
+		if( stat( longname, &statbuf ) != 0) {
```

Comment:
So, I have added an Assertion if the stat() returns 0, to make sure that it is a folder.  It is working as it did before.  But, if I create a file in the place of the data folder that it tries to write logs to (MediaVPs_2014/data for instance), it doesn't seem to hit the Assertion, in fact it tries multiple times to call mkdir_recursive anyway.  And no error occurs.  This would actually be the same behavior without any changes in this ticket I think, it would try to make it, and fail silently.  In this case it just can't write a few logs, which it seems not to complain about either.  If it were a players data folder, that would be a bigger issue I imagine, but it doesn't seem there are any checks in the entire create directory code path to ensure this happens successfully.  I think my Assertion isn't reached, because if the data file exists where there should be a folder, stat() does not find it, since the path includes a trailing /.  With the trailing /, I don't think stat will match to a file, and only a folder, so it never even hits the Assertion when the rogue file exists there.

To fix this, I would probably need to do some change to make the path sent to stat() not include a trailing slash.  The effort and complexity added here to prevent something that hasn't been an issue feels out of the scope of this PR, which is simply to stop the log messages from occurring, which it was doing fine without the Assertion I just added.  So I think I'd prefer to just get rid of the Assertion and keep this one simple.  It isn't going to be introducing any issues it seems.  If the directory exists, it skips trying to create it.  If it's a file, it still tries to create it, just like it would have before my change.  Haven't had any issues with that behavior before so I'm not sure it needs to be fixed now.